assignments/lambda-classes.js

```diff
@@ -1 +1,155 @@
 // CODE here for your Lambda Classes
+//=============PERSON CLASS | BASE CLASS=============
+class Person {
+  constructor(options){
+    this.name = options.name;
+    this.age = options.age;
+    this.location = options.location;
+    this.gender = options.gender;
+  }
+  speak() {
+    return `Hello my name is ${this.name}, I am from ${this.location}.`;
+  }
+}
+
+//=============INSTRUCTOR CLASS=============
+class Instructor extends Person {
+  constructor(instructorOptions){
+    super(instructorOptions);
+    this.specialty = instructorOptions.specialty;
+    this.favLanguage = instructorOptions.favLanguage;
+    this.catchPhrase = instructorOptions.catchPhrase;
+  }
+  demo(subject) {
+    return `Today we are learning about ${subject}` ;
+  }
+  grade(student, subject) {
+    return `${student.name} receives a perfect score on ${subject}` ;
+  }
+  gradeAdd(student){
+    const grade = Math.round(Math.random(100) * 100);
+    student.grade += grade;
+    student.assignmentCounter += 1;
+    return `${student.name}'s grade is ${student.grade} after scoring a grade of ${grade} on the assignment`;
+  }
+  gradeSubtract(student){
+    const grade = Math.round(Math.random(100) * 100);
+    student.grade -= Math.round(Math.random(100) * 100);
+    student.assignmentCounter += 1;
+    return `${student.name}'s grade is ${student.grade} after scoring a grade of ${grade} on the assignment`;
+  }
+}
+
+//=============STUDENT CLASS=============
+class Student extends Person {
+  constructor(studentOptions){
+    super(studentOptions);
+    this.previousBackground = studentOptions.previousBackground;
+    this.className = studentOptions.className;
+    this.favSubjects = studentOptions.favSubjects;
+    this.grade = studentOptions.grade;
+    this.assignmentCounter = 0;
+  }
+  listSubjects() {
+    return `${this.favSubjects}`;
+  }
+  PRAssignment(subject) {
+    return `${this.name} has submitted a PR for ${subject}`;
+  }
+  sprintChallenge(subject) {
+    return `${this.name} has begun sprint challenge on ${subject}`;
+  }
+  graduate() {
+    if (this.grade > 70) {
+        return `${this.name} has graduated with a grade of ${this.grade}!`;
```

Comment:
Currently if the grade is less than 70, this function returns undefined. Try to build in a case in to handle in the event the student cannot graduate.